FordLabs/retroquest
# Datemigration

## Overview
RetroQuest currently uses tinyblob for the following MySQL date columns:
- `date_created` in Team, Board, Feedback tables
- `last_login_date` in Team table

The date information stored in the tinyblobs are only read/writable using JPA and Hibernate, so we cannot use a MySQL migration script to update them to a DATE or TIMESTAMP column type. This limitation hinders the ability to move to SpringBoot 2.x/JDK11+.

### Notes
Most of the code in this PR will be rolled back once the JDK11 branch is merged. There will temporarily be 2 date columns in the aforementioned tables. This should not effect usage in any way.

## Testing Instructions
How to test this PR

- start from a clean MySQL db on the develop branch with the dockerdb profile
- create a board in the ui
- add a few thoughts
- click end retro
- add a Feedback
- log out
- log back in
- change to the datemigration branch and add the spring profile, datemigration
- startup the app
- check things in the MySQL database (Team, Board, Feedback)

Everything should look good, and the ui should still function the same as it is using the old column. All would be date modifications will also modify the date2 columns until we do away with the second column.
